chain/car.go

```diff
@@ -106,6 +108,43 @@ func Export(ctx context.Context, headTS block.TipSet, cr carChainReader, mr carM
 	return nil
 }
 
+func exportAMTSignedMessages(ctx context.Context, out io.Writer, smsgs []*types.SignedMessage) error {
+	ms := carWritingMessageStore(out)
+
+	cids, err := ms.storeSignedMessages(smsgs)
+	if err != nil {
+		return err
+	}
+
+	_, err = ms.storeAMTCids(ctx, cids)
+	return err
+}
+
+func exportAMTUnsignedMessages(ctx context.Context, out io.Writer, umsgs []*types.UnsignedMessage) error {
+	ms := carWritingMessageStore(out)
+
+	cids, err := ms.storeUnsignedMessages(umsgs)
+	if err != nil {
+		return err
+	}
+
+	_, err = ms.storeAMTCids(ctx, cids)
+	return err
+}
+
+func exportAMTReceipts(ctx context.Context, out io.Writer, receipts []*types.MessageReceipt) error {
+	ms := carWritingMessageStore(out)
+
+	_, err := ms.StoreReceipts(ctx, receipts)
+	return err
+}
+
+func carWritingMessageStore(out io.Writer) *MessageStore {
+	d := datastore.NewMapDatastore()
+	bs := blockstore.NewBlockstore(d)
```

Comment:
This is pretty confusing in this context. This will accumulate the entire thing in memory, unnecessarily. I think the `carExportBlockstore` should be write-only; the read methods should panic (or return error). Then it won't need this in-memory store at all.

If that doesn't work, using a MessageStore to intermediate this export is probably too indirect a path.